docs/src/pages/demos/tooltips/CustomizedTooltips.tsx

```diff
@@ -0,0 +1,198 @@
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+
+function arrowGenerator(color: string) {
+  return {
+    '&[x-placement*="bottom"] $arrow': {
+      top: 0,
+      left: 0,
+      marginTop: '-0.95em',
+      width: '3em',
+      height: '1em',
+      '&::before': {
+        borderWidth: '0 1em 1em 1em',
+        borderColor: `transparent transparent ${color} transparent`,
+      },
+    },
+    '&[x-placement*="top"] $arrow': {
+      bottom: 0,
+      left: 0,
+      marginBottom: '-0.95em',
+      width: '3em',
+      height: '1em',
+      '&::before': {
+        borderWidth: '1em 1em 0 1em',
+        borderColor: `${color} transparent transparent transparent`,
+      },
+    },
+    '&[x-placement*="right"] $arrow': {
+      left: 0,
+      marginLeft: '-0.95em',
+      height: '3em',
+      width: '1em',
+      '&::before': {
+        borderWidth: '1em 1em 1em 0',
+        borderColor: `transparent ${color} transparent transparent`,
+      },
+    },
+    '&[x-placement*="left"] $arrow': {
+      right: 0,
+      marginRight: '-0.95em',
+      height: '3em',
+      width: '1em',
+      '&::before': {
+        borderWidth: '1em 0 1em 1em',
+        borderColor: `transparent transparent transparent ${color}`,
+      },
+    },
+  };
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    button: {
+      margin: theme.spacing(1),
+    },
+    lightTooltip: {
+      backgroundColor: theme.palette.common.white,
+      color: 'rgba(0, 0, 0, 0.87)',
+      boxShadow: theme.shadows[1],
+      fontSize: 11,
+    },
+    arrowPopper: arrowGenerator(theme.palette.grey[700]),
+    arrow: {
+      position: 'absolute',
+      fontSize: 6,
+      width: '3em',
+      height: '3em',
+      '&::before': {
+        content: '""',
+        margin: 'auto',
+        display: 'block',
+        width: 0,
+        height: 0,
+        borderStyle: 'solid',
+      },
+    },
+    bootstrapPopper: arrowGenerator(theme.palette.common.black),
+    bootstrapTooltip: {
+      backgroundColor: theme.palette.common.black,
+    },
+    bootstrapPlacementLeft: {
+      margin: '0 8px',
+    },
+    bootstrapPlacementRight: {
+      margin: '0 8px',
+    },
+    bootstrapPlacementTop: {
+      margin: '8px 0',
+    },
+    bootstrapPlacementBottom: {
+      margin: '8px 0',
+    },
+    htmlPopper: arrowGenerator('#dadde9'),
+    htmlTooltip: {
+      backgroundColor: '#f5f5f9',
+      color: 'rgba(0, 0, 0, 0.87)',
+      maxWidth: 220,
+      fontSize: theme.typography.pxToRem(12),
+      border: '1px solid #dadde9',
+      '& b': {
+        fontWeight: theme.typography.fontWeightMedium,
+      },
+    },
+  }),
+);
+
+function CustomizedTooltips() {
+  const classes = useStyles();
+  const [arrowRef, setArrowRef] = React.useState<HTMLSpanElement | null>(null);
```

Comment:
@Dudrie Try implementing the same feature with 
```suggestion
  const arrowRef = React.useRef<HTMLSpanElement>(null);
```

You will probably need quite a bit of additional code to handle re-rendering. 

The current implementation is a smart way to avoid creating ref callbacks and `forceUpdate` methods.

Reply:
Oh right, `useRef` does not trigger a re-render if it's `current` changes. Using this hook would probably add more boilerplate code.

Reply:
Honestly it's one of the reasons I don't like the `ref` suffix. When used right it doesn't add information but if used wrong it adds confusion. Simply using `arrowNode` would've been clearer and then it would also be obvious that `setArrowNode` should only be used in `ref` props for host or ref forwarding components.

But there's no reason to change this now. Discussions regarding this haven't been very helpful to either side anyway.

Reply:
I forgot to comment on the `ref` suffix  situation last time. For me I don’t think the ref suffix is needed most of the time. I guess I would use it when passing something to the `ref` prop.